pkg/sql/opt/norm/general_funcs.go

```diff
@@ -137,6 +137,21 @@ func (c *CustomFuncs) IsConstJSON(expr opt.ScalarExpr) bool {
 	return false
 }
 
+// IsNullScalar returns true if the input scalar expression always evaluates to NULL.
+func (c *CustomFuncs) IsNullScalar(scalar opt.ScalarExpr) bool {
+	_, ok := scalar.(*memo.NullExpr)
+	return ok
+}
+
+func (c *CustomFuncs) IsTupleScalar(scalar opt.ScalarExpr) bool {
+	_, ok := scalar.(*memo.TupleExpr)
+	return ok
+}
```

Comment:
We need to check the type of the expression, not the type of the AST node. `IsInt` is a good example to follow. Something like:

```go
	return scalar.DataType().Family() == types.TupleFamily
```